lib/src/ui/screens/checkup/checkup_loaded_body.dart

```
@@ -19,18 +22,38 @@ class _CheckupLoadedBodyState extends State<CheckupLoadedBody> {
   CheckupStep currentStep = steps[0];
 
   void _saveCurrentLocation(CheckupStateInProgress checkupState) async {
-    final Geolocator geolocator = Geolocator();
-
+    bool showLocationFallback = true;
+    String postalCode;
     try {
-      final Position position = await geolocator.getCurrentPosition(
-        locationPermissionLevel: GeolocationPermission.locationWhenInUse,
-      );
-      final List<Placemark> places = await geolocator.placemarkFromCoordinates(
-        position.latitude,
-        position.longitude,
-      );
-      final String postalCode = places[0].postalCode;
-
+      final Geolocator geolocator = Geolocator();
+      final GeolocationStatus permission =
+          await geolocator.checkGeolocationPermissionStatus(
+              locationPermission: GeolocationPermission.locationWhenInUse);
+      if (permission == GeolocationStatus.granted ||
+          permission == GeolocationStatus.restricted) {
+        final Position position = await geolocator.getCurrentPosition(
+          locationPermissionLevel: GeolocationPermission.locationWhenInUse,
+        );
+        final List<Placemark> places =
+            await geolocator.placemarkFromCoordinates(
+          position.latitude,
+          position.longitude,
+        );
+        final String postalCode = places[0].postalCode;
+        showLocationFallback = false;
+      }
+    } catch (e) {
```

Do we know the types of the exceptions that Geolocator can throw in the case where we're lacking permissions? If so, we should tighten up this catch clause to only catch those exceptions so we don't accidentally swallow an unexpected exception.

---

It is looking like `PlatformException` is the main thing to catch here: https://github.com/Baseflow/flutter-geolocator/blob/ed70492e0d2f39efe4b4a9d345faa87bf7c620a6/lib/geolocator.dart#L215

---

Note that that's not the only exception that can be thrown, at least you can also get `PlatformException(ERROR_GEOCODING_COORDINATES, grpc failed, null)` which is bubbling up from somewhere deeper. That's also a `PlatformException` that can be thrown when we have location permissions but can't resolve `Placemark`s. The docs for Geolocator unfortunately don't indicate what kinds of errors can occur.

I left it as a catch everything since that's what was there before, I don't love catching anything so I would definitely be on board to only catch `PlatformException` I was just concerned it might throw something else, and in this case since we have a fallback it seemed better not to crash over it. ¯\_(ツ)_/¯

---

That seems reasonable.